walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,34 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.lang.String;
 
 public final class FindMeetingQuery {
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> attendees = request.getAttendees();
+
+        if (attendees.isEmpty()){
+            return Arrays.asList(TimeRange.WHOLE_DAY);
+        }
+
+        if (request.getDuration() >= (TimeRange.WHOLE_DAY.duration() + 1)){
+            return Arrays.asList();
+        }
+
+    Collection<TimeRange> availibility = Arrays.asList();
+    for (Event event: events){
+        TimeRange before = TimeRange.fromStartEnd(TimeRange.START_OF_DAY, event.getWhen().start(), false);
+        TimeRange after = TimeRange.fromStartEnd(event.getWhen().end(), TimeRange.END_OF_DAY, true);
```

Comment:
You don't use the variable 'after' it looks like you meant to add it to availibility below.

nit: s/availibility/availability

Reply:
Thanks for catching that. I fixed the indentation and the after error. Still not sure why it doesn't like this?

Reply:
It looks like Array.asList[1] is to create a fixed-size list. I think you could probably just create an ArrayLIst like (try it, not 100% sure if it will work either):

ArrayList<TimeRange> availabilites = new ArrayList<TimeRange>();
for (...) {
  // ...
  availabilities.add(before);
  availabilities.add(after);
}

return availabilities;

Going forward, I'd recommend checking the Java docs if you are not familiar with a Java API or its methods. And also, it would be good to provide more context about the error. Like is it a compile-time error, runtime error, what error message is it for others to better help.